forest/src/cli/sync_cmd.rs

```diff
@@ -0,0 +1,188 @@
+                        } else {
+                            height_diff = 0;
+                        }
+
+                        println!("\tBase:\t{:?}", base.unwrap_or_default());
+                        println!(
+                            "\tTarget:\t{:?} Height:\t({})",
+                            target.unwrap_or_default(),
+                            height
+                        );
+                        println!("\tHeight diff:\t{}", height_diff);
+                        println!("\tStage: {}", active_sync.stage());
+                        println!("\tHeight: {}", active_sync.epoch);
+                        if let Some(end_time) = active_sync.end {
+                            if let Some(start_time) = active_sync.start {
+                                let zero_time = get_naive_time_zero();
+
+                                if end_time == zero_time {
+                                    if start_time != zero_time {
+                                        let time_now = get_naive_time_now();
+                                        println!(
+                                            "\tElapsed: {:?}\n",
+                                            time_now.signed_duration_since(start_time)
+                                        );
+                                    }
+                                } else {
+                                    println!(
+                                        "\tElapsed: {:?}\n",
+                                        end_time.signed_duration_since(start_time)
+                                    );
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            SyncCommand::Wait {} => {
+                loop {
+                    // If not done syncing or runs into a error stop waiting
+                    if sync_wait(&mut client).await.unwrap_or(true) {
+                        break;
+                    }
+                }
+            }
+
+            SyncCommand::MarkBad { block_cid } => {
+                mark_bad(&mut client, block_cid)
+                    .await
+                    .map_err(stringify_rpc_err)
+                    .unwrap();
+                println!("Successfully marked block as bad");
+            }
+
+            SyncCommand::CheckBad { block_cid } => {
+                let statement = check_bad(&mut client, block_cid.clone())
+                    .await
+                    .map_err(stringify_rpc_err)
+                    .unwrap();
+                println!("Block {} is \"{}\"", block_cid, statement);
+            }
+            SyncCommand::Submit { gossip_block } => {
+                submit_block(&mut client, gossip_block)
+                    .await
+                    .map_err(stringify_rpc_err)
+                    .unwrap();
+                println!("Successfully submitted block");
```

Comment:
this doesn't fail in any way when the request should fail (running without a node running), can you look into this?

Reply:
Yeah for RPC calls that dont really return any values, we shouldnt use () as the return value because those calls will always succeed for some reason. Solution I found was using a Option<()> as the return type